src/cpp/client.cpp

```diff
@@ -63,11 +73,17 @@ Client::~Client()
         _redis = NULL;
     }
     _redis_server = NULL;
+
+    Logger& logger = Logger::get_instance();
```

Comment:
I feel like this style implementation just screams "code churn" even though I know it's effective.

Do you prefer having to add these logEnter/logExit blocks all over or would a decorator object like the following be cleaner? Please forgive my lack of c language skill

```

class LoggedClient : public IClient
   public LoggedClient(Client rootClient) { ... }

   void SomeClientMethod(...) {
      Logger& logger = Logger::get_instance();
      logger.log_data(llInfo, "some enter message");

      _rootClient.SomeClientMethod(...);

      logger.log_data(llInfo, "some exit message");
   }
}
```

This could have the added benefit of injecting the `LoggedClient` or `Client` instance and avoiding having to check every call for your log level or something like `if (loggingEnabled) { doLogging() }`

Reply:
If you look at the way we've added API tracking, it's only a single line of code added to each API point. What you're describing above duplicates the API points and involves a ton more code being written